turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2006,41 +2006,41 @@ async fn resolve_relative_request(
 
     if options_value.enable_typescript_with_output_extension {
         new_path.replace_final_constants(&|c: &RcStr| -> Option<Pattern> {
-            let result = match c.rsplit_once(".") {
-                Some((base, "js")) => Some((
+            let (base, replacement) = match c.rsplit_once(".") {
+                Some((base, "js")) => (
                     base,
                     vec![
                         Pattern::Constant(".ts".into()),
                         Pattern::Constant(".tsx".into()),
                         Pattern::Constant(".js".into()),
                     ],
-                )),
-                Some((base, "mjs")) => Some((
+                ),
+                Some((base, "mjs")) => (
                     base,
                     vec![
                         Pattern::Constant(".mts".into()),
-                        Pattern::Constant(".js".into()),
+                        Pattern::Constant(".mjs".into()),
```

Comment:
This is the fix (and the same for `cjs` below). I also refactored some of the options handling around this